inspire_dojson/hep/model.py

```diff
@@ -83,12 +83,37 @@ def convert_curated(record, blob):
     return record
 
 
+def ensure_ordered_figures(record, blob):
+    ordered_figures_dict = {}
+    unordered_figures_list = []
+
+    for figure in record.get('figures', []):
+        order = figure.get('order')
+        if order:
+            figure.pop('order')
+            ordered_figures_dict[order] = figure
+        else:
+            unordered_figures_list.append(figure)
+
+    if ordered_figures_dict:
```

Comment:
all of this `if ... else ...` ca be simplified to:
```python
record['figures'] = [value for key, value in sorted(six.iteritems(ordered_figures_dict))]
record['figures'].extend(unordered_figures_list)
```
So you skip the test, and in the case you have both ordered and unordered figures (that probably doesn't happen, but better safe than sorry), you don't throw away part of them.